pkg/apps/registry/deployconfig/strategy.go

```diff
@@ -93,6 +93,17 @@ func (strategy) PrepareForUpdate(ctx apirequest.Context, obj, old runtime.Object
 
 // Canonicalize normalizes the object after validation.
 func (strategy) Canonicalize(obj runtime.Object) {
+	dc := obj.(*appsapi.DeploymentConfig)
+
+	_, found := dc.Spec.Selector[appsapi.DeploymentConfigLabel]
+	if !found {
+		dc.Spec.Selector[appsapi.DeploymentConfigLabel] = dc.Name
```

Comment:
can dc.Spec.Selector be nil? is name always present or can we still depend on generateName running after this point?

Reply:
validation forbids empty selector
I wasn't aware generateName running after this

Reply:
generate name runs before canonicalization https://github.com/openshift/origin/blob/c07d935e1a000e89ec84dbc70683c797c1c421a8/vendor/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/registry/rest/create.go#L90-L113